hooks/open-telemetry/src/main/java/dev/openfeature/contrib/hooks/otel/OpenTelemetryHook.java

```diff
@@ -1,29 +1,69 @@
 package dev.openfeature.contrib.hooks.otel;
 
-import dev.openfeature.sdk.Client;
-import dev.openfeature.sdk.NoOpProvider;
-import dev.openfeature.sdk.OpenFeatureAPI;
+import dev.openfeature.sdk.FlagEvaluationDetails;
+import dev.openfeature.sdk.Hook;
+import dev.openfeature.sdk.HookContext;
+import io.opentelemetry.api.common.AttributeKey;
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.trace.Span;
 
-/** 
- * A placeholder.
+import java.util.Map;
+
+/**
+ * The OpenTelemetry hook  provides a way to automatically add a feature flag evaluation to a span as a span event.
+ * Refer to <a href="https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/feature-flags.md">OpenTelemetry</a>
  */
-public class OpenTelemetryHook {
-    
-    /** 
+public class OpenTelemetryHook implements Hook {
+
+    private static final String EVENT_NAME = "feature_flag";
+
+    private final AttributeKey<String> flagKeyAttributeKey = AttributeKey.stringKey(EVENT_NAME + ".flag_key");
+
+    private final AttributeKey<String> providerNameAttributeKey = AttributeKey.stringKey(EVENT_NAME + ".provider_name");
+
+    private final AttributeKey<String> variantAttributeKey = AttributeKey.stringKey(EVENT_NAME + ".variant");
+
+    /**
      * Create a new OpenTelemetryHook instance.
      */
     public OpenTelemetryHook() {
     }
 
-    /** 
-     * A test method...
+    /**
+     * Records the event in the current span after the successful flag evaluation.
      *
-     * @return {boolean}
+     * @param ctx     Information about the particular flag evaluation
+     * @param details Information about how the flag was resolved, including any resolved values.
+     * @param hints   An immutable mapping of data for users to communicate to the hooks.
      */
-    public static boolean test() {
-        OpenFeatureAPI.getInstance().setProvider(new NoOpProvider());
-        Client client = OpenFeatureAPI.getInstance().getClient();
-        return client.getBooleanValue("test2", true);
+    @Override
+    public void after(HookContext ctx, FlagEvaluationDetails details, Map hints) {
+        Span currentSpan = Span.current();
+        if (currentSpan != null) {
+            String variant = details.getVariant() != null ? details.getVariant() : String.valueOf(details.getValue());
```

Comment:
I think `String.valueOf(details.getValue()` will throw if `details.getValue()` is `null` (which is should not be, but just in case we might want to handle that.

Reply:
no, it's not. It just the return the string `null` https://docs.oracle.com/javase/7/docs/api/java/lang/String.html#valueOf(java.lang.Object)

Reply:
You're right!